android/src/main/java/com/expensify/livemarkdown/MarkdownRange.java

```diff
@@ -0,0 +1,38 @@
+package com.expensify.livemarkdown;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class MarkdownRange {
+  public final String type;
+  public final int start;
+  public final int end;
+  public final int length;
+  public final int depth;
+  public int index;
+
+  public MarkdownRange(String type, int start, int length, int depth, int index) {
+    this.type = type;
+    this.start = start;
+    this.length = length;
+    this.end = start + length;
+    this.depth = depth;
+    this.index = index;
+  }
+
+  public MarkdownRange(JSONObject range, int index) {
+    try {
+      this.type = range.getString("type");
+      this.start = range.getInt("start");
+      this.length = range.getInt("length");
+      this.end = start + length;
+      this.depth = range.optInt("depth", 1);
+      this.index = index;
```

Comment:
`index` shouldn't be a part of `MarkdownRange`.

Reply:
I optimized the current algorithm logic and we don't need `index` anymore. Fixed in https://github.com/Expensify/react-native-live-markdown/pull/534/commits/e963e97366f120ba98f52bba4229b32ae955a139